test/controllers/key_pairs_controller_test.rb

```diff
@@ -0,0 +1,59 @@
+require 'test_helper'
+
+class KeyPairsControllerTest < ActionController::TestCase
+  setup do
+    @compute_resource = FactoryGirl.create(:ec2_cr)
+  end
+
+  test "cr with key_pair should get index" do
+    get :index, {:compute_resource_id => @compute_resource.to_param}, set_session_user
+    assert_response :success
+  end
+
+  test "cr without key_pair should not get index" do
+    compute_resource = FactoryGirl.create(:libvirt_cr)
+    get :index, {:compute_resource_id => compute_resource.to_param}, set_session_user
+    assert_response :not_found
+  end
+
+  test "should download pem file" do
+    key = FactoryGirl.create(:key_pair)
+    get :download, {:compute_resource_id => @compute_resource.to_param, :id => key.id}, set_session_user
+    assert_response :success
+    assert_equal(key.secret, @response.body)
+    downloaded = Tempfile.new('tmp_pem')
```

Comment:
It gets deleted afaik ([tempfile docs](http://ruby-doc.org/stdlib-1.9.3/libdoc/tempfile/rdoc/Tempfile.html#class-Tempfile-label-Explicit+close))